openquake/hazardlib/gsim/gsim_table.py

```diff
@@ -386,20 +385,16 @@ def _setup_amplification(self, fle):
         self.amplification = AmplificationTable(fle["Amplification"],
                                                 self.m_w,
                                                 self.distances)
-        if self.amplification.element == "Sites":
-            self.REQUIRES_SITES_PARAMETERS = set(())
-            self.REQUIRES_SITES_PARAMETERS.add(self.amplification.parameter)
-        elif self.amplification.element == "Rupture":
-            # Re-set the site parameters
-            self.REQUIRES_SITES_PARAMETERS = set(())
-            self.REQUIRES_RUPTURE_PARAMETERS.add(self.amplification.parameter)
+        if self.amplification.element in ("Sites", "Rupture"):
+            self.REQUIRES_SITES_PARAMETERS = set(
+                [self.amplification.parameter])
```

Comment:
I think you are changing the behaviour here. Where the ``self.amplification.element'' corresponds to Sites then the REQUIRES_SITES_PARAMETERS is updated. When it is set to Rupture then the REQUIRES_SITES_PARAMETERS should be empty and the REQUIRES_RUPTURE_PARAMETERS updated.